experimental/grpc_attestation_client/rust/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use anyhow::{anyhow, Context};
+use oak_attestation_common::{
+    crypto::{AeadEncryptor, DiffieHellmanNegotiator},
+    report::AttestationInfo,
+};
+use oak_grpc_attestation::proto::{
+    get_attested_data_request::RequestType, get_attested_data_response::ResponseType,
+    grpc_attestation_client::GrpcAttestationClient, AttestationRequest, GetAttestedDataRequest,
+    GetAttestedDataResponse,
+};
+use tokio::sync::mpsc::Sender;
+use tonic::{transport::Channel, Request, Streaming};
+
+const MESSAGE_BUFFER_SIZE: usize = 1;
+
+/// Convenience structure for sending requests and receiving responses from the server.
+struct Client {
+    sender: Sender<GetAttestedDataRequest>,
+    response_stream: Streaming<GetAttestedDataResponse>,
+}
+
+impl Client {
+    async fn create(uri: &str) -> anyhow::Result<Self> {
+        let channel = Channel::from_shared(uri.to_string())
+            .context("Couldn't create gRPC channel")?
+            .connect()
+            .await
+            .context("Couldn't connect via gRPC channel")?;
+        let mut client = GrpcAttestationClient::new(channel);
+
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(MESSAGE_BUFFER_SIZE);
+
+        let request_stream = async_stream::stream! {
+            while let Some(message) = receiver.recv().await {
+                yield message;
+            }
+        };
```

Comment:
Is `async_stream` actually needed here? It seems it's just passing the messages through, there must be a way that only uses very basic types? In any case, probably worth encapsulating into a separate struct / fn. In fact, isn't this similar to the one that is on the server, but implemented in a completely different way?